kie-server-parent/kie-server-router/kie-server-router-proxy/src/main/java/org/kie/server/router/identity/FileStoredIdentityManager.java

```diff
@@ -0,0 +1,115 @@
+import org.jboss.logging.Logger;
+
+public class FileStoredIdentityManager implements IdentityService {
+
+    protected static final Logger log = Logger.getLogger(FileStoredIdentityManager.class);
+
+    private File userFile;
+
+    public FileStoredIdentityManager() {
+        userFile = new File("users.properties");
+        log.infof("Using <%1$s> as user repository", userFile.getAbsolutePath());
+
+    }
+
+    @Override
+    public Account verify(Account account) {
+        return account;
+    }
+  
+    @Override
+    public Account verify(Credential credential) {
+        return null;
+    }
+  
+    @Override
+    public Account verify(String id, Credential credential) {
+        if (!(credential instanceof PasswordCredential)) {
+            return null;
+        }
+        PasswordCredential passwordCredential = (PasswordCredential) credential;
+        KieServerInstanceAccount account = getAccount(id);
+        String challenge = HashUtil.hash(id, new String(passwordCredential.getPassword()));
+        if (account != null && challenge.equals(account.getHash())) {
+            log.infof("Authentication succesfully for %1$2", id);
+            return account;
+        }
+        log.infof("Authentication failed for %1$2", id);
+        return null;
+    }
+
+    private KieServerInstanceAccount getAccount(String id) {
+        Map<String, KieServerInstanceAccount> users = new HashMap<>();
+        Properties file = loadKieServerInstanceData();
+        file.stringPropertyNames().forEach(name -> users.put(name, new KieServerInstanceAccount(name, file.getProperty(name))));
+        return users.get(id);
+    }
+
+    @Override
+    public void addKieServerInstance(String kieServerInstanceId, String password) throws Exception {
+        Properties data = loadKieServerInstanceData();
+        data.put(kieServerInstanceId, HashUtil.hash(kieServerInstanceId, password));
+        saveKieServerInstanceData(data);
+    }
+
+    @Override
+    public void removeKieServerInstance(String kieServerInstanceId) {
+        Properties data = loadKieServerInstanceData();
+        data.remove(kieServerInstanceId);
+        saveKieServerInstanceData(data);
+    }
+
+    private Properties loadKieServerInstanceData() {
+        Properties properties = new Properties();
+        try (InputStream is = new FileInputStream(userFile)) {
+            properties.load(is);
+        } catch (FileNotFoundException e) {
+            log.infof("file not found %1$s", userFile);
+        } catch (IOException e) {
+            log.errorf(e, "somo error ocurred during access to %1$s", userFile);
+        }
+        return properties;
+    }
+
+    private void saveKieServerInstanceData(Properties data) {
+        try (OutputStream out = new FileOutputStream(userFile)) {
+            data.store(out, "Kie Server Router");
+        } catch (FileNotFoundException e) {
+            log.infof("file not found %1$s", userFile);
```

Comment:
FileOutputStream has a different usage of FileNotFoundException than FileInputStream. Here I would propose deleting specific handling of FileNotFoundException and left just IOException, stating that file cannot be created and logging stacktrace.
```
Throws:
FileNotFoundException - if the file exists but is a directory rather than a regular file, does not exist but cannot be created, or cannot be opened for any other reason
```

Reply:
ok